backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/CreateOvaCommand.java

```diff
@@ -229,24 +230,24 @@ private void packOva() {
         setSucceeded(true);
     }
 
-    private String getTpmData() {
+    private SecretValue<String> getTpmData() {
         var tpmDataAndHash = vmDao.getTpmData(getParameters().getEntityId());
         return tpmDataAndHash != null ? tpmDataAndHash.getFirst() : null;
     }
 
-    private String getNvramData() {
+    private SecretValue<String> getNvramData() {
         var nvramDataAndHash = vmDao.getNvramData(getParameters().getEntityId());
         return nvramDataAndHash != null ? nvramDataAndHash.getFirst() : null;
     }
 
-    private Map<VmExternalDataKind, String> getVmExternalData() {
-        Map<VmExternalDataKind, String> externalData = new HashMap<>();
-        String tpmData = getTpmData();
-        if (!StringUtils.isEmpty(tpmData)) {
+    private Map<VmExternalDataKind, SecretValue<String>> getVmExternalData() {
+        Map<VmExternalDataKind, SecretValue<String>> externalData = new HashMap<>();
+        SecretValue<String> tpmData = getTpmData();
+        if (!StringUtils.isEmpty(tpmData.getValue())) {
```

Comment:
It would be nice, I think, to have `SecretValue#isEmpty()` method.

Reply:
I introduced SecretValue.isNull that is used now (hopefully) wherever applicable.  Better than tracing what can be null or not.

As for SecretValue#isEmpty, AFAIK there is no concept of a general empty Object in Java, so there is no good way to introduce such a method.  I keep using StringUtils.isEmpty here for this reason.